model/src/main/java/com/epam/esm/entity/GiftCertificate.java

```diff
@@ -0,0 +1,45 @@
+package com.epam.esm.entity;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.validation.Valid;
+import javax.validation.constraints.*;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class GiftCertificate implements Entity {
+
+  private long id;
```

Comment:
It is bad practice to use auto-increment. Could you rework it to a better variant of id generation?
For the next time please prepare information and tell me:
1. What variants do we have and what pros-cons have each of database id generation?